app/views/student/form.php

```diff
@@ -35,4 +614,66 @@
       <input type="submit" name="edit" value="Update Profile" />
     </form>
   </div>
-</panel>
\ No newline at end of file
+</panel>
+
+<script type="text/javascript">
+function educationValueChanged() {
+  if($('.undergraduate').is(":checked"))
+  {
+    $("#uDegreeChooser").prop('required', true);
+    $("#uYearChooser").prop('required', true);
+    $("#uGraduationMonthChooser").prop('required', true);
+    $("#uGraduationYearChooser").prop('required', true);
+    $("#gDegreeChooser").prop('required', false);
+    $("#gYearChooser").prop('required', false);
+    $("#gGraduationMonthChooser").prop('required', false);
+    $("#gGraduationYearChooser").prop('required', false);
+    $(".undergraduateShow").show();
+    $(".graduateShow").hide();
+  }
+  else
+  {
+    $("#uDegreeChooser").prop('required', false);
+    $("#uYearChooser").prop('required', false);
+    $("#uGraduationMonthChooser").prop('required', false);
+    $("#uGraduationYearChooser").prop('required', false);
+    $("#gDegreeChooser").prop('required', true);
+    $("#gYearChooser").prop('required', true);
+    $("#gGraduationMonthChooser").prop('required', true);
+    $("#gGraduationYearChooser").prop('required', true);
+    $(".graduateShow").show();
+    $(".undergraduateShow").hide();
+  }
+}
+
+function lookingForValueChanged() {
+    if($('.internship').is(":checked") || $('.fulltime').is(":checked"))
+      $(".jobShow").show();
+    else
+      $(".jobShow").hide();
+
+    if($('.internship').is(":checked"))   
+      $(".internshipShow").show();
+    else
+      $(".internshipShow").hide();
+
+    if($('.fulltime').is(":checked"))   
+      $(".fulltimeShow").show();
+    else
+      $(".fulltimeShow").hide();
+
+    if($('.housing').is(":checked"))   
+      $(".housingShow").show();
+    else
+      $(".housingShow").hide();
+}
+
+educationValueChanged();
+lookingForValueChanged();
+
+// Restore previously selected items
+var industry = <?php echo json_encode(vget('industry')); ?>;
```

Comment:
we have a function toJSON (in utilities.php) that we use to json_encode and get rid of possible code injections. Should use that one over just this raw json_encode.